bcipy/helpers/triggers.py

```diff
@@ -145,61 +145,66 @@ def _write_triggers_from_inquiry_calibration(array: list,
 
     It writes in the following order:
         (I) presented letter, (II) targetness, (III) timestamp
     """
-
+    # write the offset and return the file
     if offset:
         # extract the letter and timing from the array
-        (letter, time) = array
-        targetness = 'offset_correction'
-        trigger_file.write('%s %s %s' % (letter, targetness, time) + "\n")
-
+        (symbol, time) = triggers
+        trigger_file.write(f'{symbol} offset_correction {time}\n')
+        return trigger_file
+
+    spelling_length = len(typed_text)
+    last_typed = typed_text[-1] if typed_text else None
+    target_symbol = copy_phrase[spelling_length - 1]
+
+    # because there is the possibility of incorrect letters and correction,
+    # we check here what is expected response.
+    if last_typed == target_symbol or not last_typed:
+        target_symbol = copy_phrase[spelling_length - 1]
     else:
-        # get relevant spelling info to determine what was and should be typed
-        spelling_length = len(typed_text)
-        last_typed = typed_text[-1] if typed_text else None
-        correct_letter = copy_text[spelling_length - 1]
-
-        # because there is the impassibility of incorrect letter and correction,
-        # we check here what is appropriate as a correct response
-        if last_typed == correct_letter:
-            correct_letter = copy_text[spelling_length]
-        else:
-            correct_letter = '<'
-
-        x = 0
-
-        for i in array:
+        # If the last typed and target do not match and this is not the
+        # symbols have been typed. The correct symbol is backspace.
+        target_symbol = '<'
 
-            # extract the letter and timing from the array
-            (letter, time) = i
+    for trigger in triggers:
 
-            # determine what the triggers are:
-            #       assumes there is no target letter presentation.
-            if x == 0:
-                targetness = 'fixation'
-            elif x > 1 and correct_letter == letter:
-                targetness = 'target'
-            else:
-                targetness = 'nontarget'
+        # extract the letter and timing from the array
+        (symbol, time) = trigger
 
-            # write to the trigger_file
-            trigger_file.write('%s %s %s' % (letter, targetness, time) + "\n")
+        # catch all the internal labels and assign targetness
+        if symbol == 'calibration_trigger':
+            targetness = 'calib'
+        elif symbol == 'inquiry_preview':
+            targetness = 'preview'
+        # we write the key press + key so we check the prefix is in the symbol
+        elif 'bcipy_key_press' in symbol:
+            targetness = 'key_press'
+
+        # assign targetness to the core symbols
+        elif symbol == '+':
+            targetness = 'fixation'
+        elif target_symbol == symbol:
+            targetness = 'target'
+        else:
+            targetness = 'nontarget'
```

Comment:
We might want to factor out the symbol targetness labeling into its own helper function. Seems like it might be useful for other code.

What do you think about using a dict to store the mappings? It wouldn't capture all possible values of key press options but do we need that flexibility?

Something like:

    def symbol_targetness(symbol: str, target: str) -> str:
        if symbol == target:
            return 'target'
        labels = {
            'calibration_trigger': 'calib',
            'inquiry_preview': 'preview',
            'bcipy_key_press_space': 'key_press',
            'bcipy_key_press_escape': 'key_press',
            '+': 'fixation'
        }
        return symbol_labels.get(symbol, 'nontarget')

Reply:
I agree with both of your points. I'm not sure about the key_press though since it could be any key on the keyboard (I believe?) For the sake of release, I will make a ticket to address this further. I think we can simplify this greatly!